### ISSUE TITLE
Broken TLS session while server choose TLSv1.2 instead of TLSv1.3

### ISSUE THREAD

**Issue by strayge** [secdev/scapy#2778]
#### Brief description

Broken TLS session while parsing dump from server with disabled TLSv1.3.

Client supports both TLS 1.2 and 1.3:
```
   |   |###[ TLS Extension - Supported Versions (for ClientHello) ]###
   |   |  type      = supported_versions
   |   |  len       = 5
   |   |  versionslen= 4
   |   |  versions  = [TLS 1.3, TLS 1.2]
```
But server works only with TLS 1.2:
```
   |###[ TLS Handshake - Server Hello ]###
   |  msgtype   = server_hello
   |  msglen    = 81
   |  version   = TLS 1.2
```
Looks like TLSSession saves TLSv1.3 from `ClientHello` and do not downgraded with `ServerHello`.

If restrict client to TLSv1.2 (curl `--tls-max 1.2` option) it start works as expected.

#### Environment

-   Scapy version: `2.4.4rc2.dev30`
-   Python version: `3.7.5`
-   Operating System: `Ubuntu 18.04.3 LTS`

#### How to reproduce
[scapy_client_tls13_server_tls12.pcap](https://drive.google.com/file/d/1yy0BRSwQKVLSlqnYv9hY0W0GbgoYFCED/view)
```python3
from scapy.all import *
load_layer('tls')
pcap = rdpcap('scapy_client_tls13_server_tls12.pcap')
r1 = TLS(pcap[0].load)
r2 = TLS(pcap[1].load, tls_session=r1.tls_session.mirror())
```

#### Actual result

```python3
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-4c9f13dcdcd8> in <module>
----> 1 r2 = TLS(pcap[1].load, tls_session=r1.tls_session.mirror())

~/scapy_venv/lib/python3.7/site-packages/scapy/base_classes.py in __call__(cls, *args, **kargs)
    264                 cls = config.conf.raw_layer
    265         i = cls.__new__(cls, cls.__name__, cls.__bases__, cls.__dict__)
--> 266         i.__init__(*args, **kargs)
    267         return i
    268

~/scapy_venv/lib/python3.7/site-packages/scapy/layers/tls/record.py in __init__(self, *args, **kargs)
    302     def __init__(self, *args, **kargs):
    303         self.deciphered_len = kargs.get("deciphered_len", None)
--> 304         super(TLS, self).__init__(*args, **kargs)
    305
    306     @classmethod

~/scapy_venv/lib/python3.7/site-packages/scapy/layers/tls/session.py in __init__(self, _pkt, post_transform, _internal, _underlayer, tls_session, **fields)
    902         Packet.__init__(self, _pkt=_pkt, post_transform=post_transform,
    903                         _internal=_internal, _underlayer=_underlayer,
--> 904                         **fields)
    905
    906     def __getattr__(self, attr):

~/scapy_venv/lib/python3.7/site-packages/scapy/packet.py in __init__(self, _pkt, post_transform, _internal, _underlayer, **fields)
    156         self.sniffed_on = None
    157         if _pkt:
--> 158             self.dissect(_pkt)
    159             if not _internal:
    160                 self.dissection_done(self)

~/scapy_venv/lib/python3.7/site-packages/scapy/packet.py in dissect(self, s)
    873         s = self.pre_dissect(s)
    874
--> 875         s = self.do_dissect(s)
    876
    877         s = self.post_dissect(s)

~/scapy_venv/lib/python3.7/site-packages/scapy/packet.py in do_dissect(self, s)
    837             if not s:
    838                 break
--> 839             s, fval = f.getfield(self, s)
    840             # We need to track fields with mutable values to discard
    841             # .raw_packet_cache when needed.

~/scapy_venv/lib/python3.7/site-packages/scapy/layers/tls/record.py in getfield(self, pkt, s)
    166                 if isinstance(p, _GenericTLSSessionInheritance):
    167                     if not p.tls_session.frozen:
--> 168                         p.post_dissection_tls_session_update(raw_msg)
    169
    170                 lst.append(p)

~/scapy_venv/lib/python3.7/site-packages/scapy/layers/tls/session.py in post_dissection_tls_session_update(self, msg_str)
    924
    925     def post_dissection_tls_session_update(self, msg_str):
--> 926         self.tls_session_update(msg_str)
    927
    928     def copy(self):

~/scapy_venv/lib/python3.7/site-packages/scapy/layers/tls/handshake.py in tls_session_update(self, msg_str)
    648             s.prcs = readConnState(ciphersuite=cs_cls,
    649                                    connection_end=connection_end,
--> 650                                    tls_version=s.tls_version)
    651             if not s.middlebox_compatibility:
    652                 s.triggered_prcs_commit = True

~/scapy_venv/lib/python3.7/site-packages/scapy/layers/tls/session.py in __init__(self, **kargs)
    279 class readConnState(connState):
    280     def __init__(self, **kargs):
--> 281         connState.__init__(self, read_or_write="read", **kargs)
    282
    283

~/scapy_venv/lib/python3.7/site-packages/scapy/layers/tls/session.py in __init__(self, connection_end, read_or_write, seq_num, compression_alg, ciphersuite, tls_version)
     89             from scapy.layers.tls.crypto.suites import TLS_NULL_WITH_NULL_NULL
     90             ciphersuite = TLS_NULL_WITH_NULL_NULL
---> 91         self.ciphersuite = ciphersuite(tls_version=tls_version)
     92
     93         if not self.ciphersuite.usable:

~/scapy_venv/lib/python3.7/site-packages/scapy/layers/tls/crypto/suites.py in __init__(self, tls_version)
    140         """
    141         super(_GenericCipherSuite, self).__init__()
--> 142         if tls_version <= 0x301:
    143             self.key_block_len = self._key_block_len_v1_0
    144

TypeError: '<=' not supported between instances of 'NoneType' and 'int'
```

#### Expected result

Correct work of TLSSession (saves TLSv1.2 as `tls_version` on parsing `ServerHello`) in that case.

**Comment by strayge** [comment]
Setting `tls_session.advertised_tls_version = None` before dissect `ServerHello` packet fixed this case.  
But it possible broke `TLS13ServerHello`.  

Looks like need common `TLSServerHello` class which detect TLS version by itself.

**Comment by gpotter2** [comment]
Thanks for the report, this should be fixed in https://github.com/secdev/scapy/pull/2792. Please try it out

**Comment by strayge** [comment]
@gpotter2 thank for the fix. It works as expected.

**Comment by strayge** [comment]
Minor note:  
2nd packet still detects as `TLS13ServerHello`.
It's strange, cause server does not support 1.3 and packet does not contains any field with `TLS 1.3` values.

**Comment by gpotter2** [comment]
I saw that but we can't reasonably do anything about it. It shouldn't break anything though. Only after dissecting this packet do we know that the server did not use 1.3 and we don't want to have to recompute it. Any next packet will be properly dissected though.
We could do some tricks but that will be for another PR.

**Comment by strayge** [comment]
Yes, like I said it's minor. Otherwise fix is good.
